File: Leetcode/maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
      int n=nums.length;
      HashMap<Integer,Integer>map=new HashMap<>();
      int m=0;
      int s=0;
      int l=0;
      for(int r=0;r<n;r++){
        while(map.containsKey(nums[r])){
          int cnt=map.get(nums[l]);
          map.put(nums[l],cnt-1);
          if (map.get(nums[l])==0){
            map.remove(nums[l]);
          }
          s-=nums[l];
          l++;

        }
        map.put(nums[r],1);
        s+=nums[r];
        m=Math.max(s,m);
      }
        return m;
    }
}

File: Leetcode/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
      d={}
      l=0
      s=0
      m=0
      n=len(nums)
      for r in range(n):
        while nums[r] in d:
          d[nums[l]]-=1
          if d[nums[l]]==0:
            del d[nums[l]]
          s-=nums[l]
          l+=1
        d[nums[r]]=1
        s+=nums[r]     
        m=max(s,m)
      return m

        
